src/wallet/wallet.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkvm::dpc::prelude::*;
+
+use anyhow::Result;
+use snarkos_ledger::storage::{rocksdb::RocksDB, DataMap, Map, Storage};
+use std::path::PathBuf;
+
+/// A wallet, belonging to a specific address. Currently keeps records of all
+/// coinbases ever mined to this wallet, regardless of whether they are valid
+/// or not on the current chain.
+///
+/// The path leading to the database containing all records will be structured as
+/// {data_path}/{address}/.
+///
+/// The database will just store coinbase transactions under the block hash that
+/// they belong to.
+#[derive(Clone, Debug)]
+pub struct Wallet<N: Network> {
+    /// The address associated with this wallet.
+    address: String,
+    /// A path to a directory containing all coinbase records.
+    data_path: String,
+    /// The RocksDB containing all records for the given address.
+    db: DataMap<N::BlockHash, Transaction<N>>,
```

Comment:
future consideration: if we expect `get_blocks_mined` to be called often, it would be a good idea to cache the values loaded from the database when `Wallet` is created, and to have:
```
canon_blocks_mined: u32,
blocks_mined: u32,
```
in addition to the existing fields, which would need to be updated with `push_record` and the function that marks blocks as canon; the related RPC could then use those 2 fields instead, which would be much faster